src/btree/bt_handle.c

```diff
@@ -542,7 +542,7 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt)
      * updates aren't having their transaction ids wiped after reopening the dhandle. The runtime
      * write generation is relevant here since it should remain static across the entire run.
      */
-    btree->write_gen = WT_MAX(ckpt->write_gen, conn->base_write_gen) + 1;
+    btree->write_gen = WT_MAX(ckpt->write_gen + 1, conn->base_write_gen);
```

Comment:
This is just a mistake I made when editing the original code.

It doesn't make sense for the btree's `base_write_gen` to be initialized past the connection-wide `base_write_gen` in the regular case. It means that if you open a table and then checkpoint without writing anything, the connection-wide `base_write_gen` in the metadata will move forwards.

I don't actually understand why this is causing the errors we saw in those BFs (one is `session->alter(...)` returning `EBUSY` and the other one just seems to be a metadata visibility issue). But this does fix the errors.

I'm having trouble reproducing locally and am resorting to just grepping through the code to form a theory but there's no obvious explanation that is sticking out at the moment.

Reply:
I've realized that this fix is wrong. The first time we open a btree, it's `run_write_gen` is usually going to intitialised to `conn->base_write_gen`. Then every subsequent time that it's opened, we'll do this check:

```
if (F_ISSET(session, WT_SESSION_IMPORT) || ckpt->run_write_gen <= conn->base_write_gen)
```

And realise that `ckpt->run_write_gen == conn->base_write_gen` and assign the new `run_write_gen` as `btree->write_gen` which is NOT what we want. We want to use the same one as last time.